packages/debug-nodejs/src/node/debug-nodejs.ts

```diff
@@ -51,6 +51,8 @@ export class NodeJsDebugAdapterContribution implements DebugAdapterContribution
     provideDebugAdapterExecutable(config: DebugConfiguration): DebugAdapterExecutable {
         const program = path.join(__dirname, `../../${debugAdapterDir}/out/src/nodeDebug.js`);
         return {
+            // If debugServer is set in the config, connect to that port for debugging instead of executing program
+            debugServer: config.debugServer,
```

Comment:
I guess we have to introduce a new debug configuration.
That will be easier for user to select a needed one.
```
    readonly commonConfig = {
        type: this.debugType,
        breakpoints: { filePatterns: ['[.]js$', '[.]ts$'] }
    };

    provideDebugConfigurations = [{
        ...this.commonConfig,
        request: 'attach',
        name: 'Attach by PID',
        processId: '<Enter pid to debug>'
    },
    {
        ...this.commonConfig,
        request: 'launch',
        name: 'Connect to server',
        server: {
            host: '127.0.0.1',
            port: 4711
        }
    }];
```
If we have a server configuration we don't need to put `program` and `runtime` parameters into debug executable.

Reply:
Debug extensions in vscode don't outline different configurations for `debugServer` (that I've seen) as they aren't a user-facing configuration. Your example suggests that you are connecting to the program being debugged in a different way, which isn't the case.

This PR allows the developer to attach to the debug extension to debug it without changing how that extension communicates with the program being debugged.

Reply:
I might misunderstand something, I would like to clarify everything.
How developer will attach to the debug extension?
Which configuration should be used in this case?